components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -70,11 +68,50 @@ namespace brave {
 
 namespace {
 
+// NOTE TO READER: These headers are for partners to determine the browser is
+// Brave without creating webcompat issues by creating a new user-agent. Since
+// there are tens of millions of Brave users, this presents no privacy risk.
+// Custom headers are deprecated and new partners use the
+// navigator.brave.isBrave() JavaScript API.
+constexpr char kPartnerEaffName[] = "eaff";
+constexpr char kPartnerUpholdName[] = "uphold";
+constexpr char kPartnerGrammarlyName[] = "grammarly";
+
+const std::vector<std::string> kPartnerEaffDomains = {"eaff.com",
+                                                      "stg.eaff.com"};
+const std::vector<std::string> kPartnerUpholdDomains = {
+    "sandbox.uphold.com", "api-sandbox.uphold.com", "uphold.com",
+    "api.uphold.com"};
+const std::vector<std::string> kPartnerGrammarlyDomains = {
+    "www.grammarly.com", "grammarly.com", "static.grammarly.com",
+    "gnar.grammarly.com"};
+
 BraveReferralsService::ReferralInitializedCallback*
     g_testing_referral_initialized_callback = nullptr;
 
 base::FilePath g_promo_file_path;
 
+base::Value CreateReferralHeader(
+    const char partner_name[],
+    const std::vector<std::string>& partner_domains) {
+  base::Value headers_dict(base::Value::Type::DICTIONARY);
+  base::Value cookie_names(base::Value::Type::LIST);
+  headers_dict.SetKey("cookieNames", std::move(cookie_names));
+  base::Value domains(base::Value::Type::LIST);
+  for (const auto& header : partner_domains)
+    domains.Append(header);
+  headers_dict.SetKey("domains", std::move(domains));
+
+  constexpr double referral_header_expiration = 31536000000.0;
+  headers_dict.SetDoubleKey("expiration", referral_header_expiration);
+
+  base::Value headers_sub_dict(base::Value::Type::DICTIONARY);
+  headers_sub_dict.SetStringKey(kBravePartnerHeader, partner_name);
+  headers_dict.SetKey("headers", std::move(headers_sub_dict));
+
+  return headers_dict;
+}
+
 void DeletePromoCodeFile(const base::FilePath& promo_code_file) {
```

Comment:
Is there an easy way we can call this function on startup to clean up the now-unnecessary file that's on disk for existing clients? (for example, in `BraveReferralsService::Start()`)

Reply:
Let’s keep this PR scoped the hard-coding the headers. This is unrelated.